langworld_db_data/adders/listed_value_adder.py

```diff
@@ -89,6 +131,57 @@ def _add_to_inventory_of_listed_values(
 
         return id_of_new_value
 
+    @staticmethod
+    def _get_indices_and_their_line_numbers_in_features_listed_values(
+        rows: list[dict[str, str]],
+        feature_id: str,
+        new_value_en: str,
+        new_value_ru: str,
+    ) -> list[dict[str, int]]:
+
+        value_indices_to_inventory_line_numbers: list[dict[str, int]] = []
+        """List of dictionaries, each mapping value index to line number in features_listed_values,
+        ex. [{"index": 1, "line number": 4}, {"index": 2, "line number": 5}, {"index": 3, "line number": 6}].
+        Contains all indices and line numbers of values with given feature_id.
+        """
+
+        for i, row in enumerate(rows):
+            if row["feature_id"] != feature_id:
+                continue
+
+            if row["en"] == new_value_en or row["ru"] == new_value_ru:
+                raise ValueError(f"Row {row} already contains value you are trying to add")
```

Comment:
looks very misleading that something like this is in this function. It means that the function does more than one thing and/or does not what it says in its name

Reply:
This validation now comes before application of `_get_indices_and_their_line_numbers_for_given_feature_in_features_listed_values` method